nemo/collections/nlp/models/language_modeling/bert_lm_model.py

```diff
@@ -153,19 +155,21 @@ def validation_step(self, batch, batch_idx):
         passed in as `batch`.
         """
         input_ids, input_type_ids, input_mask, output_ids, output_mask, labels = batch
-        logits = self.forward(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask,)
+        mlm_log_probs, nsp_logits = self.forward(
+            input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask
+        )
 
-        mlm_loss = self.mlm_loss(log_probs=logits[0], labels=output_ids, output_mask=output_mask)
+        mlm_loss = self.mlm_loss(log_probs=mlm_log_probs, labels=output_ids, output_mask=output_mask)
 
         if self.only_mlm_loss:
             loss = mlm_loss
         else:
-            nsp_loss = self.nsp_loss(logits=logits[1], labels=labels)
+            nsp_loss = self.nsp_loss(logits=nsp_logits, labels=labels)
 
             loss = self.agg_loss(loss_1=mlm_loss, loss_2=nsp_loss)
-        perplexity = self.perplexity_metric(mlm_loss)
-        tensorboard_logs = {"val_loss": loss, "perplexity": perplexity}
-        return {"val_loss": loss, "log": tensorboard_logs}
+        self.validation_perplexity(logits=mlm_log_probs)
```

Comment:
so it looks like here we are passing log_probs instead of logits. Is this a bug?

Reply:
In multilabel case log probs are logits normalized via `log_probs = log(logits /  exp(logits).sum())`, so log probs are logits but logits are usually not log probs. If you apply softmax to log probs, you will have probs.